src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs

```diff
@@ -0,0 +1,294 @@
+                _inputIndex += effectiveCount;
+                return 0;
+            }
+
+            // Get the number of 4 bytes blocks to transform
+            int numBlocks = (effectiveCount + _inputIndex) / 4;
+            byte[] transformBuffer = new byte[_inputIndex + effectiveCount];
+            Buffer.BlockCopy(_inputBuffer, 0, transformBuffer, 0, _inputIndex);
+            Buffer.BlockCopy(temp, 0, transformBuffer, _inputIndex, effectiveCount);
+            _inputIndex = (effectiveCount + _inputIndex) % 4;
+            Buffer.BlockCopy(temp, effectiveCount - _inputIndex, _inputBuffer, 0, _inputIndex);
+
+            tempChar = Encoding.ASCII.GetChars(transformBuffer, 0, 4 * numBlocks);
+
+            byte[] tempBytes = Convert.FromBase64CharArray(tempChar, 0, 4 * numBlocks);
+            Buffer.BlockCopy(tempBytes, 0, outputBuffer, outputOffset, tempBytes.Length);
+            return (tempBytes.Length);
+        }
+
+        public byte[] TransformFinalBlock(byte[] inputBuffer, int inputOffset, int inputCount)
+        {
+            if (inputBuffer == null) throw new ArgumentNullException(nameof(inputBuffer));
+            if (inputOffset < 0) throw new ArgumentOutOfRangeException(nameof(inputOffset), SR.ArgumentOutOfRange_NeedNonNegNum);
+            if (inputCount < 0 || (inputCount > inputBuffer.Length)) throw new ArgumentException(SR.Argument_InvalidValue);
+            if ((inputBuffer.Length - inputCount) < inputOffset) throw new ArgumentException(SR.Argument_InvalidOffLen);
+
+            if (_inputBuffer == null) throw new ObjectDisposedException(null, SR.ObjectDisposed_Generic);
+
+            byte[] temp = new byte[inputCount];
+            char[] tempChar;
+            int effectiveCount;
+
+            if (_whitespaces == FromBase64TransformMode.IgnoreWhiteSpaces)
+            {
+                temp = DiscardWhiteSpaces(inputBuffer, inputOffset, inputCount);
+                effectiveCount = temp.Length;
+            }
+            else
+            {
+                Buffer.BlockCopy(inputBuffer, inputOffset, temp, 0, inputCount);
+                effectiveCount = inputCount;
+            }
+
+            if (effectiveCount + _inputIndex < 4)
+            {
+                Reset();
+                return (Array.Empty<byte>());
+            }
+
+            // Get the number of 4 bytes blocks to transform
+            int numBlocks = (effectiveCount + _inputIndex) / 4;
+            byte[] transformBuffer = new byte[_inputIndex + effectiveCount];
+            Buffer.BlockCopy(_inputBuffer, 0, transformBuffer, 0, _inputIndex);
+            Buffer.BlockCopy(temp, 0, transformBuffer, _inputIndex, effectiveCount);
+            _inputIndex = (effectiveCount + _inputIndex) % 4;
+            Buffer.BlockCopy(temp, effectiveCount - _inputIndex, _inputBuffer, 0, _inputIndex);
+
+            tempChar = Encoding.ASCII.GetChars(transformBuffer, 0, 4 * numBlocks);
```

Comment:
I'm pretty sure that this ignores anything in excess of being mod4, so TransformFinalBlock(Encoding.ASCII.GetBytes("abc")) => empty array.  This is the same code as Desktop; but we should ensure we have tests for these boundary conditions just so we don't introduce surprises later.